Technico/Services/UserService.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Technico.Models;
+using Technico.Repositories;
+using Technico.Dtos;
+
+namespace Technico.Services;
+
+public class UserService
+{
+    private readonly UserRepository _userRepository;
+
+    public UserService(UserRepository userRepository)
+    {
+        _userRepository = userRepository;
+    }
+
+    public async Task<UserResponseDTO?> CreateAsync(UserRequestDTO createDto)
+    {
+        var users = await _userRepository.GetAllAsync();
+
+        bool VATexists = users.Any(u => u.VATNumber == createDto.VATNumber);
+        if (VATexists)return null;
+
+        bool emailExists = users.Any(u => u.Email == createDto.Email);
+        if (emailExists) return null;
+
+        var user = new User
+        {
+            VATNumber = createDto.VATNumber,
+            Name = createDto.Name,
+            Surname = createDto.Surname,
+            Address = createDto.Address,
+            PhoneNumber = createDto.PhoneNumber,
+            Email = createDto.Email,
+            Password = createDto.Password,
+        };
+        var result = await _userRepository.CreateAsync(user);
+
+        return result == null ? null : new UserResponseDTO
+        {
+            Id = result.Id,
+            Name = result.Name,
+            Email = result.Email
+        };
+    }
+
+
+    public async Task<UserResponseDTO?> UpdateAsync(Guid id, UserRequestDTO user)
```

Comment:
Ensure proper validation when updating the Role property in UpdateAsync to avoid unintended privilege escalation or data integrity issues.